scalafix-sbt/src/main/scala/scalafix/internal/sbt/CliWrapperPlugin.scala

```diff
@@ -29,15 +32,30 @@ object CliWrapperPlugin extends AutoPlugin {
       taskKey[HasMain]("Classloaded instance of main")
   }
   import autoImport._
+  private val cachedScalafixMain =
+    util.Collections.synchronizedMap(
+      new util.HashMap[(Seq[URL], String), HasMain]())
```

Comment:
prefer `String` to `URL` if you rely on hashcodes or equals working. It hits the network.

I have a refined type for this, but I'm guessing you can't depend on Refined :smile:  https://gitlab.com/fommil/drone-dynamic-agents/blob/master/src/main/scala/http/client/AsciiUrl.scala

Reply:
yeah, but make sure you canonicalise. And catch the exceptions.

Incidentally, one of the biggest code cleanup efforts in ensime was ensuring we did this everywhere we could be given a `File`. Hence, https://github.com/ensime/ensime-server/blob/3.0/api/src/main/scala/org/ensime/io/Canon.scala